### PR TITLE
Add AWS ELB (Classic and V2) to Asset Inventory 

### PR DESCRIPTION
**Pull request by kubasobon**
Add AWS ELB (Classic and V2) to Asset Inventory 

### Screenshot/Data

<details><summary>Two ELBs, Classic and V2, in AWS Console</summary>
Screenshot 2024-06-05 at 14 26 43
</details>

<details><summary>Same ELBs in Asset Inventory index</summary>
Screenshot 2024-06-05 at 13 55 31
</details>

<details><summary>Proof V2 Listeners were collected and appended to the ELB entry</summary>
Screenshot 2024-06-05 at 13 55 53
</details>

### Related Issues
Closes https://github.com/elastic/security-team/issues/9088

### Checklist
- [x] I have added tests that prove my fix is effective or that my feature works
- ~~I have added the necessary README/documentation (if appropriate)~~

#### Introducing a new rule?
no

### CODE REVIEW

File: internal/inventory/awsfetcher/fetcher_elb.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+	defer f.logger.Infof("Fetching %s - Finished", resourceName)
+
+	awsResources, err := function(ctx)
+	if err != nil {
+		f.logger.Errorf("Could not fetch %s: %v", resourceName, err)
+		return
+	}
+
+	for _, item := range awsResources {
+		assetChannel <- inventory.NewAssetEvent(
```
Comment (romulets) [inline internal/inventory/awsfetcher/fetcher_elb.go:82]: Question: can't we have a more detailed networking field?

I think somehow Elastic Load Balancers would have more things about network. Like it's the VPC, Subnet, Id? things like that

File: internal/resources/providers/awslib/elb/load_balancer.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	awsAccount   string
+	region       string
+}
+
+func (v ElasticLoadBalancerInfo) GetResourceArn() string {
+	id := pointers.Deref(v.LoadBalancer.LoadBalancerName)
+	if id == "" {
+		return ""
+	}
+	return fmt.Sprintf("arn:aws:elasticloadbalancing:%s:%s:loadbalancer/%s", v.region, v.awsAccount, id)
```
Comment (romulets) [inline internal/resources/providers/awslib/elb/load_balancer.go:40]: Nice :)

File: internal/inventory/awsfetcher/fetcher_elb.go (hunk 1 above)
Comment (kubasobon) [reply internal/inventory/awsfetcher/fetcher_elb.go:82]: Not a bad idea. WDYT about extending `inventory.AssetNetwork` with stuff like VPC ID, Subnets, Security Groups, etc.?

Comment (romulets) [reply internal/inventory/awsfetcher/fetcher_elb.go:82]: I think it's a good moment to do so! We have multiple assets that share network setups. We have better knowledge to extend `inventory.AssetNetwork`

**Pull request merged**